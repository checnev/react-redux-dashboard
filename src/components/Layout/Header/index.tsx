import React from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import {
  AppBar,
  Toolbar,
  Typography,
  IconButton,
} from '@material-ui/core';
import {
  Menu as MenuIcon,
  Brightness4,
  Brightness7,
} from '@material-ui/icons';


const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  iconChangeTheme: {},
}));

interface HeaderProps {
  openDrawer?: () => void;
  changeTheme?: () => void;
  themeMode: 'dark' | 'light';
}

const Header: React.FC<HeaderProps> = React.memo(
  ({ openDrawer, changeTheme, themeMode }: HeaderProps) => {
  const classes = useStyles();

  return (
    <AppBar position="relative" className={classes.root}>
      <Toolbar>
        <IconButton
          edge="start"
          className={classes.menuButton}
          color="inherit"
          aria-label="menu"
          onClick={openDrawer}
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h6" className={classes.title}>
          CarShare
        </Typography>
        <IconButton
          aria-label="change theme"
          onClick={changeTheme}
          color="inherit"
        >
          {themeMode === 'light' ? <Brightness4 /> : <Brightness7 />}
        </IconButton>
      </Toolbar>
    </AppBar>
  );
}
);

export default Header;
