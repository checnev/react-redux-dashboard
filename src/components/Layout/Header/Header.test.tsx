import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import Header from 'components/Layout/Header';

let component: any = null;
beforeEach(() => {
  component = document.createElement('div');
  document.body.append(component);
});

afterEach(() => {
  unmountComponentAtNode(component);
  component.remove();
  component = null;
});

const openDrawer = jest.fn();
const changeTheme = jest.fn();

describe('Header Layout', () => {
  const setUp = (props: any) => <Header {...props} />;

  it('should render render light theme', () => {
    act(() => {
      render(setUp({ openDrawer, changeTheme, themeMode: 'light'}), component);
    });

    expect(
      pretty(component.innerHTML)
    ).toMatchSnapshot();
  });

  it('should render render dark theme', () => {
    act(() => {
      render(setUp({ openDrawer, changeTheme, themeMode: 'dark' }), component);
    });

    expect(
      pretty(component.innerHTML)
    ).toMatchSnapshot();
  });

  it('should call changeTheme()', () => {
    act(() => {
      render(setUp({ openDrawer, changeTheme, themeMode: 'light' }), component);
    });

    component.querySelector('[aria-label="change theme"]').dispatchEvent(
      new MouseEvent("click", { bubbles: true })
    );

    expect(changeTheme).toHaveBeenCalledTimes(1);
  });

  it('should call openDrawer()', () => {
    act(() => {
      render(setUp({ openDrawer, changeTheme, themeMode: 'light' }), component);
    });

    component.querySelector('[aria-label="menu"]').dispatchEvent(
      new MouseEvent("click", { bubbles: true })
    );

    expect(openDrawer).toHaveBeenCalledTimes(1);
  });
});
