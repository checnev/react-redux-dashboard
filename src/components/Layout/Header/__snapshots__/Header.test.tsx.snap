// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Layout should render render dark theme 1`] = `
"<header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary makeStyles-root-5 MuiPaper-elevation4\\">
  <div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-6 MuiIconButton-colorInherit MuiIconButton-edgeStart\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"menu\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button>
    <h6 class=\\"MuiTypography-root makeStyles-title-7 MuiTypography-h6\\">CarShare</h6><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"change theme\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button>
  </div>
</header>"
`;

exports[`Header Layout should render render light theme 1`] = `
"<header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary makeStyles-root-1 MuiPaper-elevation4\\">
  <div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"menu\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button>
    <h6 class=\\"MuiTypography-root makeStyles-title-3 MuiTypography-h6\\">CarShare</h6><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"change theme\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button>
  </div>
</header>"
`;
